packages/ripple-binary-codec/tools/generateDefinitions.js

```diff
@@ -1,130 +1,150 @@
-if (process.argv.length != 3 && process.argv.length != 4) {
-  console.error(
-    'Usage: ' +
-      process.argv[0] +
-      ' ' +
-      process.argv[1] +
-      ' path/to/rippled [path/to/pipe/to]',
-  )
-  process.exit(1)
-}
-
-////////////////////////////////////////////////////////////////////////
-//  Get all necessary files from rippled
-////////////////////////////////////////////////////////////////////////
-const path = require('path')
-
-const fs = require('fs')
-function readFile(filename) {
-  try {
-    return fs.readFileSync(filename).toString('utf-8')
-  } catch (err) {
-    console.error(`Error reading file ${filename}:`, err.message)
+;(async () => {
+  if (process.argv.length != 3 && process.argv.length != 4) {
+    console.error(
+      'Usage: ' +
+        process.argv[0] +
+        ' ' +
+        process.argv[1] +
+        ' path/to/rippled [path/to/pipe/to]',
+    )
     process.exit(1)
   }
-}
-
-const sfieldHeaderFile = readFile(
-  path.join(process.argv[2], '/include/xrpl/protocol/SField.h'),
-)
-const sfieldMacroFile = readFile(
-  path.join(process.argv[2], '/include/xrpl/protocol/detail/sfields.macro'),
-)
-const ledgerFormatsMacroFile = readFile(
-  path.join(
+
+  const fetchFileFromGithub = async (repo, filename) => {
+    if (!repo.includes('tree/')) repo += '/refs/heads/develop'
+    let url = repo.replace('github.com', 'raw.githubusercontent.com')
+    url = url.replace('tree/', '')
+    url = url + '/' + filename
+
+    const response = await fetch(url)
+    return response.text()
+  }
+
+  ////////////////////////////////////////////////////////////////////////
+  //  Get all necessary files from rippled
+  ////////////////////////////////////////////////////////////////////////
+  const path = require('path')
+
+  const fs = require('fs')
+  async function readFile(repo, filename) {
```

Comment:
Please change the name of the `repo` variable, because it doesn't represent a repo when fetching from a file